web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+          settings={settings}
+          rowIndex={rows.length}
+          setSize={children?.length || 0}
+          posInSet={index}
+          isExpanded={isExpanded}
+          onCollapse={onCollapse}
+          onChange={onChange}
+        />
+      );
+
+      children?.forEach((child, index) => {
+        rows.push(
+          <DeviceRow
+            key={child.name}
+            device={child}
+            settings={settings}
+            rowIndex={rows.length}
+            level={2}
+            posInSet={index}
+            isHidden={!isExpanded}
+            onCollapse={onCollapse}
+            onChange={onChange}
+          />
+        );
+      });
+    });
+
+    return rows;
+  };
+
+  return (
+    <Table isTreeTable variant="compact" aria-label={_("Actions to find space")}>
+      <Thead>
+        <Tr>
+          <Th>{columnNames.device}</Th>
+          <Th textCenter>{columnNames.content}</Th>
+          <Th textCenter>{columnNames.size}</Th>
+          <Th textCenter>{columnNames.action}</Th>
+        </Tr>
+      </Thead>
+      <Tbody>{renderRows()}</Tbody>
+    </Table>
+  );
+};
+
+/**
+ * Widget to allow user picking desired policy to make space
+ * @component
+ *
+ * @param {object} props
+ * @param {SpacePolicy} props.currentPolicy
+ * @param {(policy: string) => void} [props.onChange]
+ */
+const SpacePolicyPicker = ({ currentPolicy, onChange = noop }) => {
+  return (
+    <OptionsPicker>
+      {SPACE_POLICIES.map((policy) => {
+        return (
+          <OptionsPicker.Option
+            key={policy.name}
+            title={policy.label}
+            body={policy.description}
+            onClick={() => onChange(policy.name)}
+            isSelected={currentPolicy?.name === policy.name}
+          />
+        );
+      })}
+    </OptionsPicker>
+  );
+};
+
+/**
+ * Section for configuring the space policy.
+ * @component
+ *
+ * @param {ProposalSettings} settings
+ * @param {boolean} [isLoading=false]
+ * @param {(settings: ProposalSettings) => void} [onChange]
+ */
+export default function ProposalSpacePolicySection({
+  settings,
+  isLoading = false,
+  onChange = noop
+}) {
+  const changeSpacePolicy = (policy) => {
+    onChange({ spacePolicy: policy });
+  };
+
+  const changeSpaceActions = (spaceAction) => {
+    const spaceActions = settings.spaceActions.filter(a => a.device !== spaceAction.device);
+    if (spaceAction.action !== "keep") spaceActions.push(spaceAction);
+
+    onChange({ spaceActions });
+  };
+
+  const currentPolicy = SPACE_POLICIES.find(p => p.name === settings.spacePolicy) || SPACE_POLICIES[0];
+
+  return (
+    <Section title={_("Find Space")} className="flex-stack">
+
+      <If
+        condition={isLoading && settings.spacePolicy === undefined}
+        then={<SectionSkeleton numRows={4} />}
+        else={
+          <>
+            <p>
+              {_("Indicate how to make free space in the selected disks for allocating the file systems:")}
+            </p>
```

Comment:
Note for myself: I start feeling that core/Section should re-introduce the `description` or `helpText` prop for this kind of _inline help_. We will see.